Makefile

```diff
@@ -677,6 +677,10 @@ ifdef ROCKSDBTESTS_END
         SUBSET := $(shell echo $(SUBSET) | sed 's/$(ROCKSDBTESTS_END).*//')
 endif
 
+ifdef ROCKSDBTESTS_ONLY
+        SUBSET := $(filter $(ROCKSDBTESTS_ONLY), $(SUBSET))
```

Comment:
To me it doesn't make sense for SUBSET and ROCKSDBTESTS_ONLY to be separate variables. As implemented, if I write

    make ROCKSDBTESTS_ONLY=does_not_exist_test check_some

it would trivially succeed. Without this change,

    $ make SUBSET=does_not_exist_test check_some
    ...
    make: *** No rule to make target `does_not_exist_test', needed by `check_some'.  Stop.

Which I think is a superior behavior. Note that for `SUBSET=does_not_exist_test make check_some` to work, the two alternate initial assignments to SUBSET would need to use `?=` instead of `:=`.

Renaming SUBSET to be RocksDB-specific is good by me.

Reply:
Is there a better way to do some of this work rather than directly int he Makefile?  Can the test rule that does parallel tests be separated into its own script (for example) so that the script could be used by both check and check_some and  CMake tests and so on...

Reply:
>Which I think is a superior behavior. Note that for SUBSET=does_not_exist_test make check_some to work, the two alternate initial assignments to SUBSET would need to use ?= instead of :=.
>
> Renaming SUBSET to be RocksDB-specific is good by me.

Sounds good, I will make this change.

> Is there a better way to do some of this work rather than directly int he Makefile? Can the test rule that does parallel tests be separated into its own script (for example) so that the script could be used by both check and check_some and CMake tests and so on...

Probably. I was also recently wanting `check_some` to use the same parallelism as when the same tests are run under `check`. So I think what you're suggesting would be quite useful.

Reply:
> > Which I think is a superior behavior. Note that for SUBSET=does_not_exist_test make check_some to work, the two alternate initial assignments to SUBSET would need to use ?= instead of :=.
> > Renaming SUBSET to be RocksDB-specific is good by me.
> 
> Sounds good, I will make this change.

Done.